source/lac/petsc_precondition.cc

```diff
@@ -488,32 +490,62 @@ namespace PETScWrappers
     AssertThrow (ierr == 0, ExcPETScError(ierr));
 
     if (additional_data.output_details)
-      PetscOptionsSetValue("-pc_hypre_boomeramg_print_statistics","1");
+#if PETSC_VERSION_GT(3,7,0)
+      PetscOptionsSetValue(NULL, "-pc_hypre_boomeramg_print_statistics","1");
 
-    PetscOptionsSetValue("-pc_hypre_boomeramg_agg_nl",
+    PetscOptionsSetValue(NULL, "-pc_hypre_boomeramg_agg_nl",
```

Comment:
we do use `PetscOptionsSetValue()` in unit tests
```
$ ag --cpp "PetscOptionsSetValue"
source/lac/petsc_precondition.cc
491:      PetscOptionsSetValue("-pc_hypre_boomeramg_print_statistics","1");
493:    PetscOptionsSetValue("-pc_hypre_boomeramg_agg_nl",
500:    PetscOptionsSetValue("-pc_hypre_boomeramg_max_row_sum", ssStream.str().c_str());
504:    PetscOptionsSetValue("-pc_hypre_boomeramg_strong_threshold", ssStream.str().c_str());
508:        PetscOptionsSetValue("-pc_hypre_boomeramg_relax_type_up", "symmetric-SOR/Jacobi");
509:        PetscOptionsSetValue("-pc_hypre_boomeramg_relax_type_down", "symmetric-SOR/Jacobi");
510:        PetscOptionsSetValue("-pc_hypre_boomeramg_relax_type_coarse", "Gaussian-elimination");
514:        PetscOptionsSetValue("-pc_hypre_boomeramg_relax_type_up", "SOR/Jacobi");
515:        PetscOptionsSetValue("-pc_hypre_boomeramg_relax_type_down", "SOR/Jacobi");
516:        PetscOptionsSetValue("-pc_hypre_boomeramg_relax_type_coarse", "Gaussian-elimination");
595:      PetscOptionsSetValue("-pc_hypre_parasails_logging","1");
629:    PetscOptionsSetValue("-pc_hypre_parasails_sym",ssStream.str().c_str());
631:    PetscOptionsSetValue("-pc_hypre_parasails_nlevels",
638:    PetscOptionsSetValue("-pc_hypre_parasails_thresh", ssStream.str().c_str());
642:    PetscOptionsSetValue("-pc_hypre_parasails_filter", ssStream.str().c_str());

tests/slepc/solve_04.cc
136:    PetscOptionsSetValue("-st_ksp_type","cg");
137:    PetscOptionsSetValue("-st_pc_type", "jacobi");
138:    PetscOptionsSetValue("-st_ksp_max_it", "10");

tests/slepc/solve_01.cc
113:    PetscOptionsSetValue("-st_ksp_type","cg");
114:    PetscOptionsSetValue("-st_pc_type", "jacobi");
115:    PetscOptionsSetValue("-st_ksp_tol", "1e-15");
132:    PetscOptionsSetValue("-st_ksp_type","preonly");
144:    PetscOptionsSetValue("-st_ksp_type","cg");
145:    PetscOptionsSetValue("-st_ksp_max_it", "10");
```
Could you please also update it there in a separate commit?